### PR TITLE
Preparations for the tracker wheel on the Google SERP

### PR DESCRIPTION
**Pull request by philipp-classen**
Preparations to use the privacy score from WhoTracks.me and show it in the Google search page. The UI is still missing. Also, it will only work on google.com.

refs  #32

### CODE REVIEW

File: src/worker/wtm-privacy-score.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
...1,0,0,0,0],"goldtokens.net":[2,0,6,0,2,2,0,0,0,0,1,2,0],"videotron.com":[8,4,12,0,2,3,1,2,0,0,6,2,1],"theroot.com":[22,1,6,0,0,2,0,0,1,0,6,2,2],"sxypix.com":[0,0,1,0,0,1,2,0,0,0,0,0,0],"ntu.edu.sg":[2,2,6,0,0,2,1,0,2,0,3,0,0],"boursier.com":[17,0,2,0,0,3,1,0,0,0,4,0,0],"ilcorsaronero.link":[2,0,4,0,0,1,1,0,1,0,1,0,1],"thefarside.com":[14,0,2,1,0,2,0,0,0,0,4,0,0],"sinonimos.com.br":[2,0,5,0,0,1,1,0,0,0,4,0,0],"zalando.no":[4,0,3,0,1,1,0,1,1,0,1,0,0],"mcmaster.com":[0,0,2,0,1,0,0,0,0,0,0,0,0],"tezfiles.com":[1,0,3,0,0,1,1,0,0,0,2,0,0],"androidpolice.com":[39,2,5,1,1,1,0,1,1,0,3,2,0],"seterra.com":[25,1,5,0,0,1,1,0,0,0,2,0,0],"wallhere.com":[2,0,2,0,0,0,1,0,0,0,1,0,0],"mejortorrentes.com":[1,0,3,0,0,1,1,0,0,0,1,0,1],"sonos.com":[13,1,6,0,4,5,1,2,0,0,8,2,1],"petitesannonces.ch":[2,0,0,0,0,0,1,0,0,0,1,0,0],"ursa-tm.ru":[3,1,7,0,0,1,1,0,0,0,3,5,0],"pcgarage.ro":[5,2,3,0,1,1,1,0,0,0,1,1,1],"streamdungeons.com":[0,1,8,0,2,1,1,0,1,0,2,0,0],"yourdictionary.com":[18,0,3,0,1,3,1,1,0,0,2,1,0],"vipboxtv.se":[4,1,2,0,1,1,1,1,0,0,1,0,3],"webfail.com":[53,1,8,0,2,4,1,1,1,0,1,2,0],"cbs.com":[29,0,9,0,1,4,1,0,2,0,10,0,0],"google.co.il":[1,1,5,0,0,0,0,1,0,0,1,0,0],"uworld.com":[4,1,6,0,1,2,1,1,1,0,5,2
```
Comment (chrmod) [inline src/worker/wtm-privacy-score.js:16]: categories should already by loaded in `storage`

Comment (chrmod) [inline src/worker/wtm-privacy-score.js:15]: can we load this from json?

File: src/manifest.json [hunk 2]
```diff
@@ -40,6 +40,13 @@
 				"vendor/@cliqz/adblocker-webextension-cosmetics/adblocker.umd.min.js",
 				"content_scripts/cosmetics.js"
 			]
+		},
+		{
+			"matches": ["*://www.google.com/*", "*://google.com/*"],
+			"run_at": "document_start",
```
Comment (chrmod) [inline src/manifest.json:46]: maybe better to load at document_end or idle? so we can skip on the even listener?

File: src/manifest.json [hunk 3]
```diff
@@ -40,6 +40,13 @@
 				"vendor/@cliqz/adblocker-webextension-cosmetics/adblocker.umd.min.js",
 				"content_scripts/cosmetics.js"
 			]
+		},
+		{
+			"matches": ["*://www.google.com/*", "*://google.com/*"],
```
Comment (chrmod) [inline src/manifest.json:45]: what about `.de` and others?

**Review by chrmod — CHANGES_REQUESTED** [review]

File: src/worker/wtm-privacy-score.js (hunk 1 above)
Comment (philipp-classen) [reply src/worker/wtm-privacy-score.js:15]: I think it is better to ship it as code, as the browser should be quicker to reprocess it. The file will not change between releases and should be cached in some from of processed bytecode.

Parsing it as JSON is preferred if you either receive it from a web request when it is the first time that the browser sees it. You by design, you have to do the work all over again on each extension/service worker start. That is my understanding.

File: src/manifest.json (hunk 3 above)
Comment (philipp-classen) [reply src/manifest.json:45]: Yes, they are currently missing. Would like like that:

https://github.com/Rob--W/dont-track-me-google/blob/abffd08f6e0d6eacd42b3ff0ccda1b6838788949/manifest.json#L14

File: src/worker/wtm-privacy-score.js (hunk 1 above)
Comment (philipp-classen) [reply src/worker/wtm-privacy-score.js:16]: That file is from the generator (other project). I would recommend to only use "lookupWtmPrivacyScoreForSite" in this file, the request is implementation detail.

It should match the categories that we have though. The order is not guaranteed to be the same.

File: src/manifest.json (hunk 2 above)
Comment (philipp-classen) [reply src/manifest.json]: Works, changed.

File: src/manifest.json (hunk 3 above)
Comment (philipp-classen) [reply src/manifest.json]: Changed.

**Pull request merged**